Bugzilla/Template.pm

```diff
@@ -227,10 +227,20 @@ sub renderComment {
     else {
         # We intentionally disable all html tags. Users should use markdown syntax.
         # This prevents things like inline styles on anchor tags, which otherwise would be valid.
+        # Note that this does not affect the html regex replacements that happens before this
+        # if block. See the comment about encoding hacks near the beginning of the function.
         $text =~ s/([<])/&lt;/g;
 
         # As a preference, we opt into all new line breaks being rendered as a new line.
         $text =~ s/(\r?\n)/  $1/g;
+
+        # Render text as markdown
+        $text = decode('UTF-8', Bugzilla->markdown_parser->render_html($text));
```

Comment:
@kyoshino discovered a problem here: https://github.com/mozilla-bteam/bmo/pull/713#discussion_r209383992.

The root of the problem is that the markdown parser (with the autolink extension enabled) doesn't respect existing anchor tags containing links. Try this out in `re.pl`.
```
$link = '<a href="https://mozilla.org">https://mozilla.org</a>';
$result = Bugzilla->markdown_parser->render_html($link);
// <p><a href="https://mozilla.org"><a href="https://mozilla.org">https://mozilla.org</a></a></p>
```
As you see, the output is an anchor nested within an anchor. This is invalid html and browsers interpt this by extracting one of the anchors and making them sibling elements, leaving one with an empty body.

When `https://mozilla.org` was found in the text before this PR, our full bug link regex had already converted it to a valid anchor tag with the text being the link itself, and then the markdown parser converted the text of the anchor tag into another anchor tag :(

In this PR, by moving the rendering of markdown before the encoding substitution stuff (line 280) we fix this problem. This is because the markdown parser doesn't have a chance to see the full link (since it is replaced with the encoding substitution). It is like we're telling the parser that we'll take care of this link ;)